rajawali/src/main/java/c/org/rajawali3d/core/ACoreComponent.java

```diff
@@ -0,0 +1,160 @@
+package c.org.rajawali3d.core;
+
+import c.org.rajawali3d.annotations.RenderThread;
+
+import android.support.annotation.CallSuper;
+import android.support.annotation.FloatRange;
+
+import net.jcip.annotations.GuardedBy;
+
+import java.util.LinkedList;
+import java.util.Queue;
+
+/**
+ * Base class for all core components; enables common render thread tasking
+ *
+ * @author Randy Picolet
+ */
+
+abstract class ACoreComponent implements CoreComponent {
```

Comment:
<tl;dr>
On Hungarian notation and naming conventions

I never liked encoding data types in variable names; it is almost always unreadable, cognitively distracting, a maintenance headache, and rarely useful as fundamental types (number vs string vs boolean vs reference) can usually be inferred and Java is type safe anyway. I've rarely seen it in Java code,

I've never seen data return types encoded in Java method names (maybe in JNI? I forget).

On the other hand, role/purpose indicators go to the semantics of an identifier, and are especially useful in reducing name collisions and lengths.

While I can live without the m or s prefix on instance/class member field names, I prefer to use them, just as I prefer to use all caps/underscores for static finals. I don't really like using `this.var = var;` all the time in constructors and setters (it is too easy to forget to use "this" later on in the method), and knowing at a glance the role of a variable - whether it's a local or parameter versus an instance member or static/class member - is simply more critical to me than its declaration details (data type, visibility, and other modifiers); it helps me know _where_ I need to go to find those when I care. I understand that IDEs visually provide role information via color and font, and full declaration info on hovers etc, but when looking at code outside of an IDE like here in github, it actually requires you to manually search through files for it (especially troublesome are inherited fields, if you don't even know it's a field) and without indicators in the name you don't get a clue about where to start.  And even in a IDE there is no standard anyway; one might use blue, another green. one bold, one italic, one underscores for different things, not to mention dev customizations.

With type names, you have the basic semantic concept (the name itself, a critical choice to me), and multiple aspects of that concept's definition and refinement in both interface and class hierarchies to deal with. Removing all shorthand indicators of interface vs abstract implementation vs concrete implementation doesn't seem like an improvement to me.  It's already hard enough to provide clear distinctions consistently and efficiently. Name collisions will increase without some kind of disambiguation for these different roles.  A and I seem simple, and in particular A reads well in English. I suppose we could use full words - AbstractFoo or BaseFoo or SomeFoo, but why? I and A are short, clear, and unambiguous de facto standards. They are not data types, but core aspects of the identification of the named artifacts that distinguish one from another.

If we don't use those, we should come up with some functionally equivalent conventions.  

</tl:dr>